extensions/lifecycle/bucketProcessor/LifecycleBucketProcessor.js

```diff
@@ -236,14 +225,21 @@ class LifecycleBucketProcessor {
      * @return {Boolean} Whether the config should be processed
      */
     _shouldProcessConfig(config) {
-        if (config.Rules.length === 0) {
-            this._log.debug('bucket lifecycle config has no rules to process', {
-                config,
+        const rulesEnabled = config.Rules.some(rule => {
+            if (rule.Status === 'Disabled') {
+                return false;
+            }
+            return Object.keys(rule).some(key => {
+                if (!this._supportedRules.includes(key)) {
+                    return false;
+                }
+                if (!rule[key].StorageClass || !this._pausedLocations.has(rule[key].StorageClass)) {
+                    return true;
+                }
+                return false;
```

Comment:
nit: seems weird this way...

either check false explicitely when the target location is paused (e.g. try every condition to 'skip'):
```suggestion
                if (rule[key].StorageClass && this._pausedLocations.has(rule[key].StorageClass)) {
                    return false;
                }
                return true;
```

or just return this value:
```suggestion
return !rule[key].StorageClass || !this._pausedLocations.has(rule[key].StorageClass)
```